### PR TITLE
Publish data before sending notifications

### PR DESCRIPTION
**Pull request by Ndpnt**

### CODE REVIEW

File: schedule.js [hunk 1]
```diff
@@ -13,12 +13,24 @@ import CGUs from './src/index.js';
...
-    app.on('error', notifier.onApplicationError.bind(notifier));
+    const delayedVersionNotitificationsParams = [];
+
+    app.on('versionRecorded', (serviceId, type, versionId) => {
+      delayedVersionNotitificationsParams.push({ serviceId, type, versionId });
+    });
+
+    app.on('changesPublished', () => {
+      delayedVersionNotitificationsParams.forEach(({ serviceId, type, versionId }) => {
+        notifier.onVersionRecorded(serviceId, type, versionId);
```
Comment (MattiSG) [inline schedule.js:26]: > Rename the function, since it is not really a handler anymore?

File: schedule.js [hunk 2]
```diff
@@ -13,12 +13,24 @@ import CGUs from './src/index.js';
 
     const app = new CGUs();
     await app.init();
-
     const notifier = new Notifier(app.serviceDeclarations, app.documentTypes);
 
-    app.on('versionRecorded', notifier.onVersionRecorded.bind(notifier));
-    app.on('fetchingError', notifier.onDocumentFetchError.bind(notifier));
-    app.on('error', notifier.onApplicationError.bind(notifier));
+    const delayedVersionNotitificationsParams = [];
```
Comment (MattiSG) [inline schedule.js:18]:
```suggestion
    const delayedVersionNotificationsParams = [];
```

File: src/index.js [hunk 3]
```diff
@@ -40,32 +40,43 @@ export default class CGUs extends events.EventEmitter {
...
+      await Promise.all(documentTrackingPromises);
+
+      if (config.get('history.publish')) {
+        try {
+          await publish();
+          this.emit('changesPublished');
+          console.log('Changes published');
+        } catch (error) {
+          console.error(`Error when trying to publish changes: ${error}`);
+          this.emit('publicationError', error);
```
Comment (MattiSG) [inline src/index.js:74]: > Unify the order of logging & emitting events 😰

File: schedule.js (hunk 1 above)
Comment (Ndpnt) [reply schedule.js:26]: ok, in this case, I also rename the other exposed functions for consistency: 
`onVersionRecorded` -> `notifyVersionRecorded` or `versionRecorded`
`onDocumentFetchError` -> `documentFetchError` or …
`onDocumentUpdateError` -> `documentUpdateError` or …
`onApplicationError` -> `applicationError` or …
?

Comment (MattiSG) [reply schedule.js:26]: Never mind, I didn't properly take into account the file… I somehow thought we were in the notifier itself, not in the caller. Sorry, this comment is not appropriate 😞

**Pull request merged**